### ISSUE TITLE
Crossterm v0.24 compatibility

### ISSUE THREAD

**Issue by mgunyho** [Canop/crokey#14]
HI! I'm using `crossterm` version 0.24 in my app (v0.23 had a bug that affected my app). However, `crokey` uses crossterm 0.23, which makes me unable to use it.

For example, with the following `Cargo.toml`:
```toml
[package]
name = "test"
version = "0.1.0"
edition = "2021"

[dependencies]
crossterm = "0.24"
crokey = "0.4"
```
and the following `main.rs`:
```rust
use crossterm::event::KeyEvent;
use crokey::key;

fn main() {
    let my_key: KeyEvent = key!(ctrl-x);
}
```
I get the error
```
error[E0308]: mismatched types
 --> src/main.rs:5:28
  |
5 |     let my_key: KeyEvent = key!(ctrl-x);
  |                 --------   ^^^^^^^^^^^^ expected struct `crossterm::event::KeyEvent`, found struct `crokey::crossterm::event::KeyEvent`
  |                 |
  |                 expected due to this
  |
  = note: perhaps two different versions of crate `crossterm` are being used?
  = note: this error originates in the macro `key` (in Nightly builds, run with -Z macro-backtrace for more info)
```
The amazing rust compiler is exactly right: the problem is because there are two different versions of crossterm in use.

Would it be possible to release a new version of crokey with crossterm 0.24? If I clone crokey and change crossterm 0.23 -> 0.24, all tests in `cargo test` pass fine.

I can submit a patch for #13 before the new release.

**Comment by Canop** [comment]
I'll do it. More than the tests of cargo test I want to check several applications using crokey. I just struggle those days with too many issues on coming back from vacations. But I should be able to do it this week-end.

**Comment by mgunyho** [comment]
Thanks a lot! No worries, let me know if I can be of help.

**Comment by Canop** [comment]
OK. What about going to crossterm 0.25 directly ?

Note that it's not totally obvious, as some versions of crossterm, for example 0.24, aren't backward compatible (my most common breaks are on the terminal's "size" not being the same on windows depending on the version).

**Comment by Canop** [comment]
I had a deeper look at Crossterm 0.25. It fundamentally changes what events are. Events aren't Copy anymore and a KeyEvent is now bigger, with a state and kind. You can't compare keys anymore with PartialEq. Crokey will have to change a lot in order to be compatible with Crossterm 0.25.

**Comment by mgunyho** [comment]
Hmm, interesting. For my use case ([tere](https://github.com/mgunyho/tere)), I don't really need the features of crossterm 0.25, so I would be happy with 0.24 for now. But I guess in the long term it would be desirable to keep up. If I need crossterm 0.25, I can definitely help with updating crokey.

Maybe we could also ask Timon to tweak KeyEvent to make things easier for crokey (like adding PartialEq).

**Comment by Canop** [comment]
In fact there's a partialEq, it's just not really one that can use for keybinding.
I'll handle the bigger 0.25 problem in another issue: https://github.com/Canop/crokey/issues/16

In the meantime, it's possible to create a version for the 0.24.

**Comment by mgunyho** [comment]
Thanks, I would appreciate it a lot!

Maybe the version of crokey could match the version of crossterm that it's based on? (This would of course break if you want to support multiple crossterm versions using feature flags or something, but I suppose it's not worth the effort to do this for now.)

**Comment by Canop** [comment]
> Maybe the version of crokey could match the version of crossterm that it's based on?

Not really, this would break as soon as I need to have a change in crokey.

But the readme should definitely have a table like

| crokey version | crossterm version |
|---------------------|-------------------------|
|  0.4.x               |  0.23.3                   |
|  0.5.x               |  0.24.0                   |
|  0.6.x               |  0.25.1                   |

with probably additional info on why.

**Comment by mgunyho** [comment]
> this would break as soon as I need to have a change in crokey

Good point.

A table like that would be useful (maybe sorted so that the latest version is at the top). Although that might make some people think that the older versions are supported as well. Another option is to just mention the currently supported version.

**Comment by Canop** [comment]
> Another option is to just mention the currently supported version.

I use Crossterm in many applications, so I know it's not always easy to switch to a newer crossterm version.

**Comment by Canop** [comment]
I just published version 0.5, compatible with crossterm 0.24.0.
Can you please confirm it's OK for you ?
(reopening in the meantime)

**Comment by mgunyho** [comment]
Seems to work fine! I'll let you know if I encounter issues.
